### DISCUSSION TITLE
See an error while trying to run "yarn test"

### DISCUSSION THREAD

**Discussion by SolarGuy8** [katspaugh/wavesurfer.js#3537] [Q&A]
Hello everyone! I've trying to run "yarn test" - one of the part in my CI/CD flow, and I see repeatable error..

```
node_modules/wavesurfer.js/dist/player.d.ts:1:29 - error TS1005: ',' expected.

1 import EventEmitter, { type GeneralEventTypes } from './event-emitter.js';
                              ~~~~~~~~~~~~~~~~~

node_modules/wavesurfer.js/dist/wavesurfer.d.ts:1:27 - error TS1005: ',' expected.

1 import BasePlugin, { type GenericPlugin } from './base-plugin.js';
                            ~~~~~~~~~~~~~
Found 2 errors.
```

Maybe someone have same error? (TS v4, wavesurfer - 7.7.1)

**Comment by katspaugh** [comment]
You should configure your build to ignore node_modules.

**Reply by 579579** [reply]
I have already attempted to resolve the issue by configuring my tsconfig.json to exclude the node_modules directory. Additionally, I have verified that my TypeScript version is indeed above 3.8, specifically version 4.2.4. Despite these measures, I am still encountering the compilation error TS1005: ',' expected.
` error  in node_modules/.pnpm/wavesurfer.js@7.7.1/node_modules/wavesurfer.js/dist/player.d.ts:1:29

TS1005: ',' expected.
  > 1 | import EventEmitter, { type GeneralEventTypes } from './event-emitter.js';
      |                             ^^^^^^^^^^^^^^^^^
    2 | type PlayerOptions = {
    3 |     media?: HTMLMediaElement;
    4 |     mediaControls?: boolean;

 error  in node_modules/.pnpm/wavesurfer.js@7.7.1/node_modules/wavesurfer.js/dist/plugins/record.d.ts:4:27

TS1005: ',' expected.
    2 |  * Record audio from the microphone with a real-time waveform preview
    3 |  */
  > 4 | import BasePlugin, { type BasePluginEvents } from '../base-plugin.js';
      |                           ^^^^^^^^^^^^^^^^
    5 | export type RecordPluginOptions = {
    6 |     /** The MIME type to use when recording audio */
    7 |     mimeType?: MediaRecorderOptions['mimeType'];

 error  in node_modules/.pnpm/wavesurfer.js@7.7.1/node_modules/wavesurfer.js/dist/wavesurfer.d.ts:1:27

TS1005: ',' expected.
  > 1 | import BasePlugin, { type GenericPlugin } from './base-plugin.js';
      |                           ^^^^^^^^^^^^^
    2 | import * as dom from './dom.js';
    3 | import Player from './player.js';
    4 | export type WaveSurferOptions = {
`

**Reply by katspaugh** [reply]
Wahat does your wavesurfer import look like?

**Reply by 579579** [reply]
the import statements look like this:

`import WaveSurfer from 'wavesurfer.js'
import RecordPlugin from 'wavesurfer.js/dist/plugins/record'`

**Reply by 579579** [reply]
Upon review of the official TypeScript documentation, it has come to my attention that the syntax "type Modifiers on Import Names" is supported from version 4.5 onwards. The most direct resolution would be to upgrade the project dependencies to version 4.5 or higher. However, given the challenges associated with upgrading certain project dependencies, your adoption of a more compatible import approach would be immensely beneficial. I appreciate your contributions and efforts.

For reference, here is the link to the documentation: 
[type Modifiers on Import Names](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-5.html)

**Reply by katspaugh** [reply]
Feel free to open a PR to split those type imports to support earlier TS versions.

You could also try specfiying a full path for your wavesurfer imports like this:

```js
import WaveSurfer from 'wavesurfer.js/dist/wavesurfer.esm.js'
import RecordPlugin from 'wavesurfer.js/dist/plugins/record.esm.js'
```

Not sure if it would help but worth a try.
